test/unit/startup/test_bootstrap.py

```diff
@@ -536,24 +553,37 @@ def test_existing_worker_successful_restore(
         worker_persistence_info_mock.load.assert_called_once_with(config=config)
         worker_persistence_info_mock.save.assert_not_called()
         create_worker_mock.assert_not_called()
-        worker_info_mock.save.assert_not_called()
+        if get_instance_id_value is not None and saved_instance_id is None:
+            worker_info_mock.save.assert_called_once_with(config=config)
+        else:
+            worker_info_mock.save.assert_not_called()
 
+    @mark.parametrize(
+        "load_result",
+        [
+            None,
+            WorkerPersistenceInfo(worker_id=WORKER_ID, instance_id=INSTANCE_ID + "1"),
+        ],
```

Comment:
nit: what do you think about adding IDs to these parametrizations. I see you updated the docstring, but giving each one an ID might help maintainers understand more quickly. It also helps understand what broke in the test report. Something like:

```suggestion
        [
            param(None, id="no-prior-state"),
            param(WorkerPersistenceInfo(worker_id=WORKER_ID, instance_id=INSTANCE_ID + "1"), id="prior-state-diff-instance"),
        ],
```

What do you think?

Reply:
Sure, I've also added IDs it to the other modified test.